webapp/commonComponents/expression/expressionEditorPopup.js

```diff
@@ -11,15 +11,43 @@ import ExpressionNode from './nodes/expressionNode'
 
 import { useExpressionEditorPopupState, mapStateToProps } from './expressionEditorPopupState'
 
+import AdvancedExpressionEditorPopup from './advancedExpressionEditor'
+
 const ExpressionEditorPopup = props => {
-  const { nodeDefCurrent, isBoolean, variables, onChange, onClose } = props
+  // IsGeneralExpression: this not just a boolean expression, i.e. the advanced mode can be used
+  const { nodeDefCurrent, isBoolean, variables, onChange, onClose, hideAdvanced } = props
 
-  const { query, queryDraft, exprDraft, exprDraftValid, updateDraft } = useExpressionEditorPopupState(props)
+  const {
+    query,
+    queryDraft,
+    exprDraft,
+    exprDraftValid,
+    updateDraft,
+    resetDraft,
+    advanced,
+    setAdvancedEditor,
+  } = useExpressionEditorPopupState(props)
 
   const i18n = useI18n()
 
+  if (advanced)
+    return (
+      <AdvancedExpressionEditorPopup
+        setAdvancedEditor={setAdvancedEditor}
+        revertToBasicMode={() => {
+          resetDraft()
+          setAdvancedEditor(false)
+        }}
+        {...props}
+      />
+    )
+
   return (
     <Popup className="expression-editor-popup" onClose={onClose} padding={20}>
+      <button hidden={hideAdvanced} onClick={() => setAdvancedEditor(true)} style={{ position: 'absolute', right: 0 }}>
+        {i18n.t('nodeDefEdit.expressionsProp.advanced')}
+      </button>
+
```

Comment:
I would switch here the Advanced vs Basic editor. (to avoid duplicate code <Popup/>)